plugins/source/aws/resources/services/elasticsearch/vpc_endpoints_fetch.go

```diff
@@ -0,0 +1,62 @@
+package elasticsearch
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/elasticsearchservice"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client/services"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchElasticsearchVpcEndpoints(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
```

Comment:
Sorry, I thought I commented on this yesterday, but apparently not. Can the list / describe pattern here be implemented as a PreResourceResolver to better make use of the goroutine pool we have in the Plugin-SDK?

Reply:
I think this falls under https://github.com/cloudquery/plugin-sdk/issues/338.
If we use `PreResourceResolver` we will only be able to call `DescribeVpcEndpoints` with a single `vpcEndpointID`

Reply:
Ah okay I see, thanks!